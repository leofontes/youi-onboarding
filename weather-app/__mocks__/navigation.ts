export const navigationProp = {
  navigation: {
    addListener: jest.fn(),
    isFocused: jest.fn(),
    navigate: jest.fn(),
    getParam: jest.fn(),
  },
};
