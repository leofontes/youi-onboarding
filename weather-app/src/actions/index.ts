import * as tmdbActions from './tmdbActions';
import * as youtubeActions from './youtubeActions';

export {
  tmdbActions as tmdb,
  youtubeActions as youtube,
};
