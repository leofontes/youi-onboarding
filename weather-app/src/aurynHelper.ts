import { DeviceInfo, Ref } from '@youi/react-native-youi';
import { NativeModules, findNodeHandle } from 'react-native';
import { once } from 'lodash';

const systemName = DeviceInfo.getSystemName();
const { Cloud, RefUtils } = NativeModules;

interface AurynHelper {
  hasHardwareBackButton: boolean;
  isRoku: boolean;
  updateCloudScene: (component: React.RefObject<any>) => void;
  togglePointerEvents: (ref: Ref, enabled: boolean) => boolean;
  toggleParentPointerEvents: (ref: Ref, enabled: boolean) => boolean;
}

export const AurynHelper: AurynHelper = {
  hasHardwareBackButton: !['iOS'].includes(systemName),
  isRoku: Cloud.isCloudServer,
  updateCloudScene: (component) => {
    if (!Cloud.isCloudServer || !component.current) return;
    setTimeout(() => {
      Cloud.exportSubtree(findNodeHandle(component.current), false);
      Cloud.sendFocusMap();
    }, 300);
  },
  togglePointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
  toggleParentPointerEvents: once((ref: Ref, enabled: boolean) => {
    RefUtils.setParentCompositionPointerEvents(findNodeHandle(ref), enabled);
    return true;
  }),
};
