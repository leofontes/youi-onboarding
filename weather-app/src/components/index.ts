export { BackButton } from './backButton';
export { DiscoverContainer } from './discoverContainer';
export { List } from './list';
export { ListItem } from './listitem';
export { Timeline } from './timeline';
export { ToggleButton } from './toggleButton';
export { ToggleGroup } from './toggleGroup';
export { NavigationBar } from './navigationBar';
export { TvContainer } from './tvContainer';
export { Error } from './error';
export { withOrientation } from './withOrientation';
export { LiveContainer } from './liveContainer';
export { LiveListItem } from './liveListitem';
export { BackgroundImage } from './backgroundImage';
