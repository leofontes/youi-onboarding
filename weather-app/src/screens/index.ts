export { Splash } from './splash';
export { Lander } from './lander';
export { Pdp } from './pdp';
export { Search } from './search';
export { Profile } from './profile';
export { VideoScreen as Video } from './video';
export { AdOverlay } from './../components/adManager/list-item-ad';
