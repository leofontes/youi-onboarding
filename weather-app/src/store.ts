import promise from 'redux-promise-middleware';
import createDebounce from 'redux-debounced';
import thunk from 'redux-thunk';
import { rootReducer, StoreState } from './reducers';

import { configureStore, ConfigureStoreOptions, AnyAction } from '@reduxjs/toolkit';

export default configureStore({
  reducer: rootReducer,
  middleware: [createDebounce(), thunk, promise]
} as ConfigureStoreOptions<StoreState, AnyAction>);
