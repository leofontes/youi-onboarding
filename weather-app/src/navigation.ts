import * as Screens from './screens';
import { createStackNavigator, createSwitchNavigator, StackNavigatorConfig } from 'react-navigation';
import { AurynHelper } from './aurynHelper';

const stackOptions: StackNavigatorConfig = {
  headerMode: 'none',
  cardStyle: {
    backgroundColor: 'transparent',
    opacity: 1,
  },
  transitionConfig: () => ({
    transitionSpec: {
      duration: 0,
    },
    containerStyle: {
      backgroundColor: 'transparent',
    },
    screenInterpolator: () => {},
  }),
};

const createNavigator = AurynHelper.isRoku ? createSwitchNavigator : createStackNavigator;
const AppStack = createNavigator(
  {
    Lander: { screen: Screens.Lander },
    PDP: { screen: Screens.Pdp },
    Search: { screen: Screens.Search },
    Profile: { screen: Screens.Profile },
    Video: { screen: Screens.Video },
    AdOverlay: { screen: Screens.AdOverlay },
  },
  {
    ...stackOptions,
    initialRouteName: 'Lander',
  },
);

const SplashStack = createStackNavigator(
  { Splash: { screen: Screens.Splash } },
  {
    ...stackOptions,
    initialRouteName: 'Splash',
  },
);

const rootNavigationStack = createSwitchNavigator({
  Splash: SplashStack,
  App: AppStack,
});

export default rootNavigationStack;
