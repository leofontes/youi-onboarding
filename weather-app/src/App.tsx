import React from 'react';
import store from './store';
import Stack from './navigation';
import { FormFactor } from '@youi/react-native-youi';
import { Provider } from 'react-redux';
import { BackgroundImage } from './components';

export default class YiReactApp extends React.Component {
  render = () =>
    <Provider store={store}>
      <BackgroundImage source={!FormFactor.isHandset ? 'res://drawable/default/CES-auryn-10ft-bg.png' : 'res://drawable/default/Gradient-Bottom-2.png'}>
        <Stack />
      </BackgroundImage>
    </Provider>
}
