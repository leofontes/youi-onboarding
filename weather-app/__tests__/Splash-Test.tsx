
import React from 'react';
import { Splash } from '../src/screens';
import { Provider } from 'react-redux';
import { navigationProp } from '../__mocks__/navigation';
import store from '../src/store';
import renderer from 'react-test-renderer';

test('renders correctly with error', () => {
  const tree = renderer
    .create(<Provider store={store}>
      <Splash {...navigationProp} error="Splash Screen with Error"/>
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

test('renders correctly', () => {
  const tree = renderer
    .create(<Provider store={store}>
      <Splash {...navigationProp} />
    </Provider>)
    .toJSON();

  expect(tree).toMatchSnapshot();
});


