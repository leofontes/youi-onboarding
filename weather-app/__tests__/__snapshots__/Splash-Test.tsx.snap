// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#2C2C2F",
      "flex": 1,
    }
  }
>
  <Composition
    onTimelinesLoaded={[Function]}
    source="Auryn_Splash"
  >
    <TimelineRef
      loop={false}
      name="SplashIn"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
    <TimelineRef
      loop={false}
      name="SplashOut"
      onAborted={[Function]}
      onCompleted={[Function]}
      onCompletedForward={[Function]}
      onCompletedReverse={[Function]}
      onCompositionDidLoad={[Function]}
      onPaused={[Function]}
      onPausedForward={[Function]}
      onPausedReverse={[Function]}
      onPlay={[Function]}
      onPlayForward={[Function]}
      onPlayReverse={[Function]}
      onStarted={[Function]}
      onStartedForward={[Function]}
      onStartedReverse={[Function]}
    />
    <ViewRef
      name="Loader"
      onCompositionDidLoad={[Function]}
      onTimelinesLoaded={[Function]}
    >
      <TimelineRef
        autoplay={true}
        loop={true}
        name="Loop"
        onAborted={[Function]}
        onCompleted={[Function]}
        onCompletedForward={[Function]}
        onCompletedReverse={[Function]}
        onCompositionDidLoad={[Function]}
        onPaused={[Function]}
        onPausedForward={[Function]}
        onPausedReverse={[Function]}
        onPlay={[Function]}
        onPlayForward={[Function]}
        onPlayReverse={[Function]}
        onStarted={[Function]}
        onStartedForward={[Function]}
        onStartedReverse={[Function]}
      />
    </ViewRef>
  </Composition>
</View>
`;

exports[`renders correctly with error 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#143672",
      "borderColor": "black",
      "borderWidth": 3,
      "flex": 1,
      "flexWrap": "wrap",
      "justifyContent": "center",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 36,
      }
    }
  >
    Splash Screen with Error
  </Text>
</View>
`;
